KPI-team1-frontend/src/App.tsx

```diff
@@ -1,14 +1,48 @@
 import { Outlet } from "react-router-dom";
 import Sidebar from "./components/Sidebar";
+import { supabase } from "./supabase";
+import { useEffect, useState } from "react";
+
+interface User {
+  id: string;
+  email?: string;
```

Comment:
Why is email optional?

Reply:
Sorry for any confusion. Initially, I set the email type as a string. However, I encountered an error, as shown in the screenshot below, when attempting to fetch the user from Supabase:
<img width="1093" alt="Screenshot 2023-10-03 at 10 23 09" src="https://github.com/WomenPlusPlus/deploy-impact-23-kpi-1/assets/112515323/c41723ec-41e7-4bd6-892e-fc98ee84b54f">
When look at supabase, I realised that in the Supabase-provided User table, there's an option to create a user using a phone, making the email type potentially undefined in that scenario. To address this, I quickly made the email parameter optional. However, now when I review the code again, I realised I could resolve this by simply modifying the code to: `setUser({ id: user.id, email: user.email || "" });` without the need for setting the email as optional. I will update the code accordingly.